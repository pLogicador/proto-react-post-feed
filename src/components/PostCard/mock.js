export const postCardPropsMock = {
  title: "title ex1",
  body: "body ex1",
  id: 1,
  cover: "img/img.jpeg",
};
