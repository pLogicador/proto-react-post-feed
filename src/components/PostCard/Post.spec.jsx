import { render, screen } from "@testing-library/react";
import { PostCard } from ".";
import { postCardPropsMock } from "./mock";

const props = postCardPropsMock;

describe("<PostCard />", () => {
  it("should render PostCard correctly", () => {
    render(<PostCard {...props} />);

    expect(screen.getByRole("img", { name: props.title })).toHaveAttribute(
      "src",
      "img/img.jpeg"
    );
    expect(
      screen.getByRole("heading", { name: "1 title ex1" })
    ).toBeInTheDocument();
    expect(screen.getByText("body ex1")).toBeInTheDocument();
  });

  it("should match snapshot", () => {
    const { container } = render(<PostCard {...props} />);
    expect(container.firstChild).toMatchSnapshot();
  });
});
