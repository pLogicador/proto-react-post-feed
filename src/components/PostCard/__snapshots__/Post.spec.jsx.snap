// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<PostCard /> should match snapshot 1`] = `
<div
  class="post"
>
  <img
    alt="title ex1"
    src="img/img.jpeg"
  />
  <div
    class="post-content"
  >
    <h2>
      1
       
      title ex1
    </h2>
    <p>
      body ex1
    </p>
  </div>
</div>
`;
